### PR TITLE
Updates to how we save synthesis metamer / mad image over synthesis

### PR DESCRIPTION
**Pull request by billbrod**
**Describe the change in this PR at a high-level**

This PR started with an attempt to fix two small issues: 
- calling `metamer.saved_metamer` with a metamer object that had just been initialized (or had `synthesize()` called with `store_progress=False`) would raise an Error as we tried to stack an empty list
- if you called `synthesize(store_progress=i)` where `i` is an int greater than 1, and then tried to call `plot_synthesis_status` with `iteration!=None`, the resulting plot wouldn't properly line up the values from `losses` and `saved_metamer`, which could easily lead to errors.

as I started to fix these two issues, I made some slight changes to how these two attributes (and their corresponding ones in `MADCompetition`) are updated. In addition to fixing the above issues:

- `losses` will always have length `num_iter+1`, where `num_iter` is the number of iterations of synthesis undergone so far. we do this by caching a `_current_loss` attribute at the end of `synthesize` and concatenating it to `losses` in the getter. Before, `losses[-1]` was not the loss of the current metamer, because it was computed before the update.
- if `_current_loss` is not set, we try to compute the current loss. we may not be able to, e.g., just after initialization (because we don't have our initial image for synthesis), in which case we return an empty tensor.
- analogously, `saved_metamer[-1]` is always the current metamer.
- :bug: `Metamer.objective_function` now takes in `metamer` as its first object (*not* `metamer_representation = model(metamer)`, because I realized we were adding the range penalty based on the current metamer, which is probably not what you want. (Note that MADCompetition doesn't have this issue and so there's no change)

**Outstanding questions / particular feedback**

Are the user-facing descriptions (in the attribute docstrings, especially) clear enough? I myself was confused with how things were being handled, so I tried to make them clearer. In particular, I tried to make it clear how to map between loss and `saved_metamer`, and their relationship with the current metamer.

Other than docstrings, I didn't change the documentation, which I think is correct -- basically I want them to now behave "as expected". Can you try them out and let me know if I succeeded?

~Also: by adding the new `_current_loss` attribute to Metamer and MADCompetition, I have broken backwards compatibility on their save/load. Is that okay? Should I communicate that somewhere? I, in general, don't promise about being able to save/load across versions. But in this case, I could also provide a snippet of code to make old saves compatible with the new version:~

```python
import torch
old_save = torch.load("path/to/old_save.pt")
old_save["_current_loss"] = None
torch.save(old_save, "new_save.pt")
# and then load as normal
```

Sep 8 2025: For now, we handle this saving/loading for users, raising a FutureWarning and pointing to a new section in the docs about compatibility which includes the above code snippet.

**Describe changes**

In addition to above:
- Lots of tests to make this work
- Removed schematic for `MetamerCTF._check_convergence`

**Checklist**

Affirm that you have done the following:

- [x] I have described the changes in this PR, following the template above.
- [x] I have added any necessary tests.
- [x] I have added any necessary documentation. This includes docstrings, updates to existing files found in `docs/`, or (for large changes) adding new files to the `docs/` folder.

### CODE REVIEW

**Review by BalzaniEdoardo — CHANGES_REQUESTED** [review]
Looks good overall and I think the flow is clear. 
My only concern is the image retrieval: when `iteration` isn’t an exact multiple of `store_progress`, we silently display a different checkpoint. 

A lightweight fix would be to either warn or include the *actual* iteration in the figure title (e.g., “requested 999, showing 0”) when they differ. Also, please spell this out in the docstrings for `display_mad_image`, `plot_pixel_values`, etc. 

One more subtlety: the current integer-division logic always floors to the previous checkpoint, so with large steps (`store_progress=1000`) a request like `iteration=999` still shows iteration `0`. Consider selecting the closest checkpoint (e.g., `round(iteration / store_progress)`) or adding a parameter to choose the policy.

File: src/plenoptic/synthesize/synthesis.py [hunk 1]
```diff
@@ -730,9 +731,27 @@ def allowed_range(self) -> tuple[float, float]:
...
+        current_loss = self._current_loss
+        # this will happen if we haven't run synthesize() yet or got
+        # interrupted
+        if current_loss is None:
+            try:
+                # compute current loss, no need to compute gradient
+                with torch.no_grad():
+                    current_loss = self.objective_function().item()
+            except Exception:
+                # this will happen if setup() has not been called and so we can't
```
[the remark this answers is not in the document]
Comment (billbrod) [reply src/plenoptic/synthesize/synthesis.py:751]: good idea, I'll try and come up with one here

[the remark this answers is not in the document]
Comment (billbrod) [reply src/plenoptic/synthesize/synthesis.py:751]: changed how I'm handling this: I now have the `metamer` attribute return `torch.empty(0)` when it hasn't been initialized, and have `objective_function` do the same. that makes this easier to check.
